var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -230,8 +203,7 @@ def check_fortran_compiler(self):
     depends_on("rocprim", when="+rocm")
 
     # Build dependencies
-    depends_on("python@2.6:2.8", type="build", when="@:3.10")
-    depends_on("python@2.6:2.8,3.4:", type="build", when="@3.11:")
+    depends_on("python@2.6:2.8,3.4:", type="build")
```

Comment:
Should this just say python3 is required?

Reply:
Technically, it supports all versions of Python that Spack supports, so we could just get rid of all version specifications. There are thousands of packages where this is true, so I haven't bothered to update all of them.

Reply:
That actually is an interesting point, would it be worth having a variable in spack for that version range? Especially for applying to other repo versions than current it might be a nice helper to have around.

Reply:
> That actually is an interesting point, would it be worth having a variable in spack for that version range?

You mean the range of Python versions that Spack currently supports? Or the range of versions needed by this package?

> Especially for applying to other repo versions than current it might be a nice helper to have around.

You mean for external repos (not builtin), or for people using older core with newer builtin, or both? I have no idea how to support the former, and the latter isn't currently supported, although that's something we plan on doing before a 1.0 release.

Reply:
I meant the range that spack supports, in terms of "supported by the python/package.py file in the spack repo in use".  It's a property of the python package we have, that could reasonably be used by another package, but it's easy enough to end up with a version of  python in spack that's outside that range that specifying it still seems useful.  Does that make more sense?

I mainly meant for external repos, or dealing with pythons built with older repo versions or brought in as externals such that they aren't accounted for in the python package.

Reply:
> I meant the range that spack supports, in terms of "supported by the python/package.py file in the spack repo in use

Welll - I'm usually able to `spack install pkg@ver` - even though this `ver` is not listed in pkg/package.py [assuming the tarball exists - the sha256 check gets skipped]

Reply:
I have no intention nor capability of supporting external repos, it's up to the maintainers of external repos to support builtin imo.